runtime/cudaq/qis/managers/default/DefaultExecutionManager.cpp

```diff
@@ -37,60 +30,17 @@ class DefaultExecutionManager : public cudaq::BasicExecutionManager {
   nvqir::CircuitSimulator *simulator() {
     return nvqir::getCircuitSimulatorInternal();
   }
-  /// @brief To improve `qudit` allocation, we defer
-  /// single `qudit` allocation requests until the first
-  /// encountered `apply` call.
-  std::vector<cudaq::QuditInfo> requestedAllocations;
-
-  /// @brief Allocate all requested `qudits`.
-  void flushRequestedAllocations() {
-    if (requestedAllocations.empty())
-      return;
-
-    allocateQudits(requestedAllocations);
-    requestedAllocations.clear();
-  }
 
 protected:
-  void allocateQudit(const cudaq::QuditInfo &q) override {
-    requestedAllocations.emplace_back(2, q.id);
-  }
-
-  void allocateQudits(const std::vector<cudaq::QuditInfo> &qudits) override {
```

Comment:
It was useful from a performance perspective to observe (in some way) that we were really allocating a qvector, and to queue them up and request the allocation all at once (means we can create the correct GPU pointer once, and not once + N-1 changes to the data on device). Is this type of functionality retained somewhere else? maybe i'll see it further down

Reply:
It is not present anywhere else. I can implement it on the `CircuitSimulator` layer, so that JITed kernels would also benefit from it.